### PR TITLE
Update release instructions to include proper branching strategy and protect release branches

### LINKED ISSUE #214: Update release documentation to separate release branch creation from release day process
As the initial release has shown (see #213), there were a few problems with the release process originally defined in #131. Most importantly, we need to change the branching strategy a bit, to properly allow for creating the release branch ahead of time, which is necessary to not block merging of other PRs ready for `trunk`.

That release branch was already created early for the initial release, which made sense due to the above, but then following the release instructions created some problems. Also, it shouldn't be allowed to just directly commit against the release branches, they should be protected in the same way as `trunk`. In other words, PRs should also have to go against the `release/**` branches.

### PR DESCRIPTION
**Pull request by felixarntz**
## Summary

Fixes #214

## Checklist

- [x] PR has either `[Focus]` or `Infrastructure` label.
- [x] PR has a `[Type]` label.
- [x] PR has a milestone or the `no milestone` label.

### CODE REVIEW

File: docs/Releasing-the-plugin.md
Comment (dainemawer) [inline docs/Releasing-the-plugin.md]: "As usual, make sure to request a review from 2 plugin maintainers. Once approved, the pull request can be merged."

File: docs/Releasing-the-plugin.md
Comment (dainemawer) [inline docs/Releasing-the-plugin.md:69]: @felixarntz - considering the number of WP core versions - do we want to scope this? Maybe the previous 3 versions or something along those line? Just a thought!

File: docs/Releasing-the-plugin.md
Comment (dainemawer) [inline docs/Releasing-the-plugin.md:70]: Should we add some direction with regards to local environments to test this on? Like `wp-env`?

**Review by dainemawer — APPROVED** [review]
@felixarntz left a few comments, approving regardless :)

File: docs/Releasing-the-plugin.md
Comment (felixarntz) [reply docs/Releasing-the-plugin.md:69]: @dainemawer Note the "different WordPress core versions _supported_" which is very limited. The plugin will always have one of the latest WP core versions as a minimum requirement, so I don't foresee this to ever be more than 2 or 3 WP core versions.

For example, right now we would only need to test with WP 5.8 and WP 5.9.

File: docs/Releasing-the-plugin.md
Comment (felixarntz) [reply docs/Releasing-the-plugin.md:70]: @dainemawer I think that would be a bit overkill here, it can be any environment where these points can be tested. It can be a local site, somebody's production site, a quickly spun up test site using one of those throwaway site providers - so IMO specifying more details here wouldn't be necessary, but if it was, it should go elsewhere since it would go too much into detail and thus make this doc too long - it's already fairly long based on the current more crucial requirements defined.

File: docs/Releasing-the-plugin.md
Comment (dainemawer) [reply docs/Releasing-the-plugin.md:69]: Awesome :)

File: docs/Releasing-the-plugin.md
Comment (dainemawer) [reply docs/Releasing-the-plugin.md:70]: Cool beans!

**Review by JustinyAhin — APPROVED** [review]
Great additions @felixarntz.

**Pull request merged**
